.github/workflows/check.yml

```diff
@@ -12,43 +12,100 @@ jobs:
     runs-on: ubuntu-24.04
 
     steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: ${{ github.event_name == 'pull_request' && 2 || 0 }}
```

Comment:
Why do we need specifically 2 commits of history for pull requests?

Reply:
It should be explained in https://stackoverflow.com/questions/74265821.